pkg/pool-weighted/contracts/BaseWeightedPool.sol

```diff
@@ -94,8 +99,8 @@ abstract contract BaseWeightedPool is BaseMinimalSwapInfoPool, WeightedMath {
         return WeightedMath._calculateInvariant(normalizedWeights, balances);
     }
 
-    function getNormalizedWeightsAndMaxWeightIndex() external view returns (uint256[] memory, uint256) {
```

Comment:
Sorry, I wasn't clear in my previous message :sweat-smile:

I meant let's keep `getWeightsAndMaxIndex` in the base class, but in the special case of `WeightedPool`, implement it by calling `getWeights` and then also returning `maxIndex`, to keep that implementation clean.